### ISSUE TITLE
Error "Failed to allocate remote process memory for library string. If you are running an Antivirus or Secruity Suite, try temporarily disabling it." 

### ISSUE THREAD

**Issue by PartridgeProgramming** [rawleyfowler/linux-wse-installer#2]
Hello, so I followed the instructions on the readme to install this and got the error message in the title. Here is the terminal output as well:
WSE Loader by cmpxchg8b, K700, AgentSmith (Build date: Apr 11 2019 21:01:59)
OS version: 6.1.7601 SP 1.0
Command line:
WSE path: Z:\home\alden\Games\MountBlade Warband\
Game executable origin: path
Game executable path: Z:\home\alden\Games\MountBlade Warband\mb_warband.exe
Game executable checksum: 1D0AF088
Library path: Z:\home\alden\Games\MountBlade Warband\WSELib.dll
Setting library search path
kernel32.lib local address: 7B600000
SetDllDirectoryW local address: 7B61B940
Library path: Z:\home\alden\Games\MountBlade Warband\
Library path size in bytes: 79
Library path remote address: 110000
Remote thread handle: 80
00c8:err:module:import_dll Library fmodex.dll (which is needed by L"Z:\\home\\alden\\Games\\MountBlade Warband\\mb_warband.exe") not found
00c8:err:module:import_dll Library SkinMagic.dll (which is needed by L"Z:\\home\\alden\\Games\\MountBlade Warband\\mb_warband.exe") not found
MESA-INTEL: warning: Haswell Vulkan support is incomplete
00c8:err:module:LdrInitializeThunk Importing dlls for L"Z:\\home\\alden\\Games\\MountBlade Warband\\mb_warband.exe" failed, status c0000135
Exit code: C0000135
Injecting
kernel32.lib local address: 7B600000
LoadLibraryW local address: 7B60C844
WSE Loader by cmpxchg8b, K700, AgentSmith (Build date: Apr 11 2019 21:01:59)
OS version: 6.1.7601 SP 1.0
Command line:
WSE path: Z:\home\alden\Games\MountBlade Warband\
Game executable origin: path
Game executable path: Z:\home\alden\Games\MountBlade Warband\mb_warband.exe
Game executable checksum: 1D0AF088
Library path: Z:\home\alden\Games\MountBlade Warband\WSELib.dll
Setting library search path
kernel32.lib local address: 7B600000
SetDllDirectoryW local address: 7B61B940
Library path: Z:\home\alden\Games\MountBlade Warband\
Library path size in bytes: 79
Library path remote address: 110000
Remote thread handle: 80
Exit code: C0000135
Injecting
kernel32.lib local address: 7B600000
LoadLibraryW local address: 7B60C844
Failed to allocate remote process memory for library string.
If you are running an Antivirus or Security Suite, try temporarily disabling it.

I am using Linux Mint 21.3 if that helps.

**Comment by rawleyfowler** [comment]
Try running this:

```
WINEARCH="win32" WINEPREFIX="$HOME/.wine_warband" winetricks vcrun2005sp1
```

Then re-launching the game.

**Comment by PartridgeProgramming** [comment]
I ran the command and ran "wse_warband" in my linux_wse_installer folder and got the same exact error and terminal log.

Edit: I have Warband on Steam and have it set to use Proton Experimental if that helps at all.

**Comment by rawleyfowler** [comment]
This is the error, why it's happening I am not sure.

```
00c8:err:module:import_dll Library fmodex.dll (which is needed by L"Z:\home\alden\Games\MountBlade Warband\mb_warband.exe") not found
00c8:err:module:import_dll Library SkinMagic.dll (which is needed by L"Z:\home\alden\Games\MountBlade Warband\mb_warband.exe") not found
```

**Comment by PartridgeProgramming** [comment]
Okay, so I reinstalled your program and now mostly it works. The mouse cursor flickers but Warband seems to be otherwise functional.

**Comment by rawleyfowler** [comment]
@PartridgeProgramming Do you have the correct drivers installed for your machine? You *have* to have the 32 bit aka i386 versions installed as well.

**Comment by PartridgeProgramming** [comment]
How do I check if I do? That might be the key to my new issue, the game getting stuck on "loading game" when I click on new game.

**Comment by rawleyfowler** [comment]
@PartridgeProgramming Follow this guide https://github.com/lutris/docs/blob/master/InstallingDrivers.md

**Comment by PartridgeProgramming** [comment]
In the guide it says for Mint: "NOTE: Mint is not supported and these instructions WILL NOT work in Mint without manual intervention that is outside the scope of this guide. Following them WILL leave your packages in a broken state!!".

Here is what I get when I run "inxi -Gx", if it helps:
Graphics:
  Device-1: Intel Xeon E3-1200 v3/4th Gen Core Processor Integrated Graphics
    vendor: Acer Incorporated ALI driver: i915 v: kernel bus-ID: 00:02.0
  Display: x11 server: X.Org v: 1.21.1.4 driver: X: loaded: modesetting
    unloaded: fbdev,vesa gpu: i915 resolution: 1920x1080~60Hz
  OpenGL: renderer: Mesa Intel HD Graphics 4600 (HSW GT2)
    v: 4.6 Mesa 23.2.1-1ubuntu3.1~22.04.2 direct render: Yes

**Comment by rawleyfowler** [comment]
Try changing the game options to the oldest version of DirectX, I think it's 7. That has worked for similar issues in the past.

**Comment by PartridgeProgramming** [comment]
So when I run the launcher from your executable I can't seem to be able to change the directx version.
![image](https://github.com/rawleyfowler/linux-wse-installer/assets/60394309/928d3951-219f-4b18-8bc4-2c46abe1e48b)
If I run WSELoader.exe directly I can change the directx (it was on 9 before), but the game won't even boot.
![Screenshot from 2024-04-22 13-44-18](https://github.com/rawleyfowler/linux-wse-installer/assets/60394309/72c6726c-e83c-4344-ad4d-6e9339b3acf5)
![Screenshot from 2024-04-22 13-44-40](https://github.com/rawleyfowler/linux-wse-installer/assets/60394309/c70c157e-71cd-4319-a30c-21de6c0629c1)

**Comment by rawleyfowler** [comment]
Can you confirm the contents of your `wse_warband` script please?

**Comment by rawleyfowler** [comment]
You can try running this and seeing if it works:

```
WINE_PREFIX="$HOME/.wine_warband wine "/path/to/WSELoader.exe"
```

Which is what your `wse_warband` script _should_ be.

**Comment by PartridgeProgramming** [comment]
> Can you confirm the contents of your `wse_warband` script please?

Here is the contents of my wse_warband script:
#!/bin/sh
WINEPREFIX="$HOME/.wine_warband" wine "/home/alden/Games/MountBlade Warband/WSELoader.exe"

**Comment by rawleyfowler** [comment]
Have you tried running `WINEPREFIX="$HOME/.wine_warband" wine "/home/alden/Games/MountBlade Warband/WSELoader.exe"` in your current shell? I wonder if `sh` is causing issue.

**Comment by PartridgeProgramming** [comment]
When I ran that command, I got the following output on my terminal:
WSE Loader by cmpxchg8b, K700, AgentSmith (Build date: Apr 11 2019 21:01:59)
OS version: 6.1.7601 SP 1.0
Command line:
WSE path: Z:\home\alden\Games\MountBlade Warband\
Game executable origin: path
Game executable path: Z:\home\alden\Games\MountBlade Warband\mb_warband.exe
Game executable checksum: 1D0AF088
Library path: Z:\home\alden\Games\MountBlade Warband\WSELib.dll
Setting library search path
kernel32.lib local address: 7B600000
SetDllDirectoryW local address: 7B61B940
Library path: Z:\home\alden\Games\MountBlade Warband\
Library path size in bytes: 79
Library path remote address: 110000
Remote thread handle: 80
MESA-INTEL: warning: Haswell Vulkan support is incomplete
0094:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0094:err:winediag:nodrv_CreateWindow The explorer process failed to start.
0094:err:ole:apartment_createwindowifneeded CreateWindow failed with error 0
0094:err:ole:apartment_createwindowifneeded CreateWindow failed with error 0
0094:err:ole:apartment_createwindowifneeded CreateWindow failed with error 0
Exit code: 1
Set library search path
Injecting
kernel32.lib local address: 7B600000
LoadLibraryW local address: 7B60C844
Library path: Z:\home\alden\Games\MountBlade Warband\WSELib.dll
Library path size in bytes: 99
Library path remote address: 110000
Remote thread handle: 8C
Injected library address: 54C0000
Injected
alden@alden-Aspire-TC-605:~$ 00cc:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
00cc:err:winediag:nodrv_CreateWindow The explorer process failed to start.
wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
00cc:err:ntdll:RtlpWaitForCriticalSection section 02DDB0A8 "?" wait timed out in thread 00cc, blocked by 0000, retrying (60 sec)

The last line keeps repeating.
